default/AI/ProductionAI.py

```diff
@@ -203,538 +169,8 @@ def getBestShipRatings(loc=None, verbose=False):
-            if pid is None:  # TODO: is this check still necessary?
-                continue
-            species_name = ColonisationAI.empire_species_by_planet.get(pid, '')
-            try_counter = 0  # tracks how many design tries since last improved design found here
-            for _ , shipDesignID in theseDesignIDs:
-                designStats = foAI.foAIstate.get_weighted_design_stats(shipDesignID, ColonisationAI.empire_species_by_planet.get(pid, ''))
-                shipDesign = fo.getShipDesign(shipDesignID)
-                if not shipDesign.productionLocationForEmpire(empireID, pid):
-                    continue
-                cost = get_design_cost(cur_turn, shipDesign, pid)
-                # nattacks = sum( designStats.get('attacks', {1:1}).keys() )
-                old_design_rating = foAI.foAIstate.rate_psuedo_fleet( [(-1, shipDesignID, species_name)] ).get('overall', 0)
-                # TODO: determine better tactical rating adjustment for speed
-                new_design_rating = old_design_rating * (1.0 + designStats.get('tact_adj', 0.0))
-                design_rating = [old_design_rating, new_design_rating][style_index]
-                costRating = design_rating/(max(0.1, cost))
-                if verbose and ( int(old_design_rating/10) != int(new_design_rating/10) ):
-                    print "design %s (for species %s) has cost %.1f, old rating %.1f and new rating %.1f"%(
-                                        shipDesign.name(False), species_name, cost, old_design_rating, new_design_rating)
-                if (try_counter > 200) and (costRating < 0.1 * bestCostRating):
-                    break
-                if costRating > localBestCostRating:
-                    try_counter = 0
-                    localBestCostRating = costRating
-                    bestDesignID = shipDesignID
-                    bestDesign = shipDesign
-                    if verbose:
-                        print "at planet %s, new local best design %s with rating %.1f, costRating %.1f, hull %s and partslist %s"%(
-                                    ppstring(PlanetUtilsAI.planet_name_ids([pid])), shipDesign.name(False), design_rating, costRating, shipDesign.hull, list(shipDesign.parts))
-                    if costRating > bestCostRating:
-                        bestCostRating = costRating
-                else:
-                    try_counter += 1
-            if localBestCostRating > 0.0:
-                if verbose:
-                    print "\t\t adding design id %d (%s) (species %s) with costRating %.4f at pid %d" % (
-                        bestDesignID, bestDesign.name(False), species_name, localBestCostRating, pid)
-                locDetail.append( [localBestCostRating, pid, bestDesignID, bestDesign] )
-        if not locDetail:
-            return []
-        locDetail.sort(reverse=True)
-
-        # Since we haven't yet implemented a way to target military ship construction at/near particular locations
```

Comment:
This partition function type weighting approach is still being used, a few lines up, so the comments should be moved but not entirely deleted, or some later coder will again think we have horribly botched our weighting function :smiley: